src/index.ts

```diff
@@ -52,9 +53,12 @@ export type ProxyCheckResponse = {
   status: Status
   node?: string
   query_time?: string
-} & {
-  [ipAddress: string]: IPAddressInfo
-}
+} & (
+  | {
+      [ipAddress: string]: IPAddressInfo
+    }
+  | IPAddressInfo
```

Comment:
Can you confirm if the addition of type `IPAddressInfo` breaks existing versions?